content/doc/tutorials/using-jenkinsfile-runner-github-action-to-build-jenkins-pipeline.adoc

```diff
@@ -0,0 +1,308 @@
+        env:
+          - key: hello
+            value: world
+----
+* Create a Jenkinsfile:
+[source,groovy]
+----
+pipeline {
+    agent any
+    stages {
+        stage('test casc env') {
+            steps {
+                echo "JCasC env.hello: ${env.hello}"
+            }
+        }
+    }
+}
+----
+* Specify the jcasc.yml in the GitHub Actions input:
+[source,yaml]
+----
+uses:
+  jenkinsci/jfr-container-action@master
+with:
+  command: run
+  jenkinsfile: Jenkinsfile
+  pluginstxt: plugins.txt
+  jcasc: jcasc.yml
+----
+
+You can find more link:https://github.com/jenkinsci/configuration-as-code-plugin/tree/master/demos[examples] provided by configuration-as-code-plugin and learn how to configure the Jenkins instance without using the UI page. 
+Some plugins might not have concrete examples but you can debug and find their JCasC in the UI page. 
+You can check the configuration in Manage Jenkins -> Configuration as Code -> View Configuration. 
+Then you can copy the parts you need to the JCasC file.
+
+=== Add and configure some plugins (Optional)
+
+Jenkins has a powerful plugin system. 
+You can add the plugins in the plugin list file and configure the plugins in the JCasC YAML file as needed.
+
+In this example, we show how to install JDK8 in the ephemeral Jenkins instance.
+* Specify `adoptopenjdk` plugin in the plugins.txt file. 
+As the version is not specified, the latest version will be installed.
+[source,txt]
+----
+adoptopenjdk
+----
+* Create a JCasC Yaml file called jcasc.yml and specify which JDK version to install.
+[source,yaml]
+----
+tool:
+  jdk:
+    installations:
+      - name: jdk8
+        home: "~/jdk8"
+        properties:
+          - installSource:
+              installers:
+                - adoptOpenJdkInstaller:
+                    id: "jdk8u332-b09"
+----
+* Specify the jcasc.yml in the GitHub Actions input.
+[source,yaml]
+----
+uses:
+  jenkinsci/jfr-container-action@master
+with:
+  command: run
+  jenkinsfile: Jenkinsfile
+  pluginstxt: plugins.txt
+  jcasc: jcasc.yml
+----
+
+=== Configure ephemeral Jenkins instance (Optional)
+
+Sometimes JCasC might not be able to provide the configurations you need. 
+In this case, you can use link:https://www.jenkins.io/doc/book/managing/groovy-hook-scripts/[Groovy Hook Scripts] to set up the ephemeral Jenkins instance. 
+These Groovy scripts will have full access to the ephemeral Jenkins server and will be executed right after Jenkins starts up. 
+Please note this option and its core are still in progress so it’s not mentioned in the Jenkinsfile Runner GitHub Actions official guide. 
+But it does work.
+
+In this example, we show how to use the Groovy script to set up the Jenkins instance:
+* Create a directory called “groovy.init.d” (this name can be something else) to store all your Groovy setup scripts.
+* Create a Groovy file called “test.groovy”. (Please don’t name it as init.groovy because this name is already occupied.) Then, add the debug output:
```

Comment:
```suggestion
In this example, we review how to use Groovy scripts to set up the Jenkins instance:
. Create a directory, for example “groovy.init.d”, to store all your Groovy setup scripts.
. Create a Groovy file called “test.groovy”.
* Do not name it `init.groovy` because this name is already occupied.
. Then, add the debug output:
```
syntax and formatting suggestions